File: leetsync/2346-largest-3-same-digit-number-in-string/largest-3-same-digit-number-in-string.go
func largestGoodInteger(num string) string {
	n := len(num)
	maxChar := ""
	for i := 0; i < n-2; i++ {
		if num[i] == num[i+1] && num[i] == num[i+2] {
			current := num[i : i+3]
			if maxChar < current {
				maxChar = current
			}
		}
	}
	return maxChar
}

File: leetsync/2346-largest-3-same-digit-number-in-string/largest-3-same-digit-number-in-string.java
class Solution {
    public String largestGoodInteger(String num) {
        int n = num.length();
        char maxChar = ' ';
        for (int i = 2; i < n; i++) {
            if (num.charAt(i) == num.charAt(i - 1) && num.charAt(i) == num.charAt(i - 2)) {
                maxChar = (char) Math.max(maxChar, num.charAt(i));
            }
        }
        return maxChar == ' ' ? "" : new String(new char[] { maxChar, maxChar, maxChar });
    }
}